semantic-codegen/semantic-codegen.cabal

```diff
@@ -0,0 +1,82 @@
+cabal-version:       2.4
+-- Initial package description 'semantic-codegen.cabal' generated by 'cabal
+--  init'.  For further documentation, see
+-- http://haskell.org/cabal/users-guide/
+
+name:                semantic-codegen
+version:             0.1.0.0
+-- synopsis:
+-- description:
+-- bug-reports:
+license:             MIT
+license-file:        LICENSE
+author:              The Semantic Authors
+maintainer:          opensource+semantic@github.com
+copyright:           (c) 2019 GitHub, Inc.
+category:            Language
+extra-source-files:  CHANGELOG.md
+
+tested-with:         GHC == 8.6.5
+
+common haskell
+  default-language:    Haskell2010
+  ghc-options:
+    -Weverything
+    -Wno-missing-local-signatures
+    -Wno-missing-import-lists
+    -Wno-implicit-prelude
+    -Wno-safe
+    -Wno-unsafe
+    -Wno-name-shadowing
+    -Wno-monomorphism-restriction
+    -Wno-missed-specialisations
+    -Wno-all-missed-specialisations
+    -Wno-star-is-type
+  if (impl(ghc >= 8.8))
+    ghc-options: -Wno-missing-deriving-strategies
+
+library
+  import:              haskell
+  exposed-modules:     AST.Deserialize
+                       AST.GenerateSyntax
+                       AST.Grammar.TH
+                       AST.Token
+                       AST.Unmarshal
```

Comment:
I’m of two minds about this. On the one hand, I kinda think we should try to stick with the naming convention of naming the module hierarchy after the package (i.e. `Codegen.*`), on the other hand… it’s not that big a deal, file find will still locate things successfully, and it’d be annoying to have to rename these and wait for CI.

Let’s leave it as is for now at least, and try consciously not to accept this as precedent for breaking from the naming scheme in other new packages 😊